networks/archive/1b_corrected_correlation_analysis.py

```diff
@@ -0,0 +1,506 @@
+pa14_U, pa14_s, pa14_Vh = np.linalg.svd(pa14_compendium_log10, full_matrices=False)
+
+print(pao1_compendium_T.shape)
+print(pao1_U.shape, pao1_s.shape, pao1_Vh.shape)
+
+print(pa14_compendium_T.shape)
+print(pa14_U.shape, pa14_s.shape, pa14_Vh.shape)
+
+# Convert ndarray to df to use corr()
+pao1_U_df = pd.DataFrame(
+    data=pao1_U,
+    index=pao1_compendium_T.index,  # columns=pao1_compendium.index
+)
+pa14_U_df = pd.DataFrame(
+    data=pa14_U,
+    index=pa14_compendium_T.index,  # columns=pa14_compendium.index
+)
+
+# +
+# Correlation of U
+# Since `corr()` computes pairwise correlation of columns we need to invert U
+
+pao1_corr_spell = pao1_U_df.iloc[:, :num_singular_values].T.corr()
+pa14_corr_spell = pa14_U_df.iloc[:, :num_singular_values].T.corr()
+# -
+
+"""# Plot heatmap
+plt.figure(figsize=(20, 20))
+h1a = sns.clustermap(pao1_corr_spell.abs(), cmap="viridis")
+h1a.fig.suptitle(
+    f"log transform + SPELL corrected using {num_singular_values} vectors (PAO1)",
+    y=1.05,
+)"""
+
+"""plt.figure(figsize=(20, 20))
+h2a = sns.clustermap(pa14_corr_spell.abs(), cmap="viridis")
+h2a.fig.suptitle(
+    f"log transformed + SPELL corrected using {num_singular_values} vectors (PA14)",
+    y=1.05,
+)"""
+
+# ## PCA + correlation
+#
+# Here we are going to calculate the correlation of the gene coefficients, stored in the PC components matrix. We expect this correlation matrix to look very similar to the SPELL results. This is acting as a positive control that we implemented the SPELL correction correctly
+
+# +
+# Embed expression data into low dimensional space
+pca = PCA(n_components=num_PCs)
+
+model_pao1 = pca.fit(pao1_compendium)
+
+# Using components matrix: gene x PCs (each PC value is how much a gene contributes)
+print(model_pao1.components_.shape)
+pao1_pc_weights_df = pd.DataFrame(
+    data=model_pao1.components_, columns=pao1_compendium.columns
+)
+pao1_pc_weights_df.head()
+
+# +
+pca_variance = model_pao1.explained_variance_
+
+plt.figure(figsize=(8, 6))
+plt.bar(
+    range(num_PCs), pca_variance, alpha=0.5, align="center", label="individual variance"
+)
+plt.legend()
+plt.title("Variance explained for PAO1")
+plt.ylabel("Variance ratio")
+plt.xlabel("Principal components")
+plt.show()
+
+# +
+model_pa14 = pca.fit(pa14_compendium)
+
+# Using components matrix: gene x PCs (each PC value is how much a gene contributes)
+print(model_pa14.components_.shape)
+pa14_pc_weights_df = pd.DataFrame(
+    data=model_pa14.components_, columns=pa14_compendium.columns
+)
+
+# +
+pca_variance = model_pa14.explained_variance_
+
+plt.figure(figsize=(8, 6))
+plt.bar(
+    range(num_PCs), pca_variance, alpha=0.5, align="center", label="individual variance"
+)
+plt.legend()
+plt.title("Variance explained for PA14")
+plt.ylabel("Variance ratio")
+plt.xlabel("Principal components")
+plt.show()
+# -
+
+# Correlation
+pao1_corr_pca = pao1_pc_weights_df.corr()
+pa14_corr_pca = pa14_pc_weights_df.corr()
+
+"""# Plot heatmap
```

Comment:
Do you want to plot these?

Reply:
Since it takes 2-3 minutes per correlation plot and I was playing around with different variables, I was commenting out different plots. So yes, this was intended